### PR TITLE
Add custom resource decryption test

### PR DESCRIPTION
**Pull request by britzl**
Added unit test to check that it is possible to change the resource decryption at runtime

### CODE REVIEW

File: engine/resource/src/resource_archive.cpp [hunk 1]
```diff
@@ -59,7 +59,6 @@ namespace dmResourceArchive
     }
 
     void RegisterResourceDecryption(FDecryptResource decrypt_resource) {
-        assert(g_ResourceDecryption == 0);
```
Comment (britzl) [inline engine/resource/src/resource_archive.cpp:62]: This was wrong. We assign a default decryption on init and this assert prevented anyone from changing to a custom one.

File: engine/resource/src/test/test_resource_archive.cpp [hunk 2]
```diff
@@ -858,6 +858,37 @@ TEST(dmResourceArchive, LoadFromDisk_Compressed)
     dmResourceArchive::Delete(archive);
 }
 
+
+static dmResourceArchive::Result TestDecryption(void* buffer, uint32_t buffer_len)
```
Comment (britzl) [inline engine/resource/src/test/test_resource_archive.cpp:862]: Stupid test "decryption" with a hardcoded output

Comment (JCash) [inline engine/resource/src/test/test_resource_archive.cpp:862]: Looking at it now, I wonder if the path of the resource is required?
(E.g. for doing different encrypting based on the file type)
E.g. skipping encryption for larger files like textures and sounds.

**Review by JCash — APPROVED** [review]

File: engine/resource/src/test/test_resource_archive.cpp (hunk 2 above)
Comment (britzl) [reply engine/resource/src/test/test_resource_archive.cpp:862]: That is a good point and perhaps something to add as part of a new task or based on user requests!

**Pull request merged**